lightning/src/ln/channel.rs

```diff
@@ -1843,6 +1843,9 @@ impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
 			return Err((None, ChannelError::Close("Got wrong number of HTLC signatures from remote")));
 		}
 
+		// TODO: Merge these two, sadly they are currently both required to be passed separately to
```

Comment:
Sure, but just because they're intended for different uses doesn't mean we should be passing two separate vectors over with redundant data. We can pass one Vec and let ChannelMonitor figure out how to put what where.

Reply:
Okay but that means ChannelMonitor doing more repacking stuff on sensible code paths, which has been error-prone in the past (something we can improve with well-documented named structures). There is also few intersection between them beyond index

Reply:
Right, but its even easier to get confused passing things twice than it is to get confused passing things once and having to think about a well-documented value :).